pandas/io/parsers.py

```diff
@@ -1707,11 +1706,7 @@ def _convert_to_ndarrays(
                     or is_extension_array_dtype(cast_type)
                 ):
                     try:
-                        if (
-                            is_bool_dtype(cast_type)
-                            and not is_categorical_dtype(cast_type)
-                            and na_count > 0
-                        ):
+                        if not is_ea and na_count > 0 and is_bool_dtype(cast_type):
```

Comment:
do we have something that actually hits this. you changed the comparison and nothing broke here? (e.g. to check for ea and not categorical).

Reply:
This block is hit by 77 tests across io.excel and io.parser, for which 33 have `is_categorical_dtype(cast_type)` being True, all within `io.parser.test_dtypes.test_categorical_*`.

Note that `is_ea` is True whenever `is_categorical_dtype(cast_type)` is True, so what's changing here is that EA types which are not categorical are no longer raising if they are Boolean and have null values. Behavior for categorical is unchanged. I think this is appropriate because EA types are to define their own logic via `_from_sequence_of_strings` which is called within `_cast_types` that occurs immediately after this try-except.